File: 024_Swap_Nodes_In_Pairs/Solution.java
// Author: Code-Beast

// Definition for singly-linked list.
public class ListNode {
    int val;
    ListNode next;
    ListNode(int x) { val = x; }
}

// MY SOLUTION 1 (Recursion)
// Runtime: 2 ms
class Solution {
    public ListNode swapPairs(ListNode head) {
        if (head == null || head.next == null) {
            return head;
        }

        ListNode nextPair = head.next.next,
                 newHead = head.next;
        newHead.next = head;
        head.next = swapPairs(nextPair);

        return newHead;
    }
}

File: 024_Swap_Nodes_In_Pairs/Solution.py
#!python3
# Author: Code-Beast

# Definition for singly-linked list.
class ListNode:
    def __init__(self, x):
        self.val = x
        self.next = None

# MY SOLUTION 1 (Recursion)
# Runtime: 32ms
class Solution:
    def swapPairs(self, head):
        """
        :type head: ListNode
        :rtype: ListNode
        """
        if not head or not head.next:
        	return head

        nextPair = head.next.next
        newHead = head.next
        newHead.next = head
        head.next = self.swapPairs(nextPair)

        return newHead

